rhesus-monkeys/scripts/convert-meta.py

```diff
@@ -0,0 +1,89 @@
+
+# Create the adult metadata files
+    # FIO There are no additional Mother_ID in file 11
+df_metadata_maternal_bl = pl.read_csv(unzip_path / "metadata_maternal_blood.csv")
+df_metadata_maternal_ur = pl.read_csv(unzip_path / "metadata_maternal_urine.csv")
+df_metadata_maternal_pl = pl.read_csv(unzip_path / "metadata_maternal_placenta.csv")
+
+df_adult_meta1 = df_metadata_maternal_bl.select(["Mother_ID", "Mother_age", "GD_Delivery", "Group", "Mode_birth", "Reject", "Infant_sex"]).rename({"Mother_age": "Age_at_conception"
+, "GD_Delivery": "gestation_day_at_delivery", "Group": "obesity_classification", "Mode_birth": "mode_of_birth"}).clean_names()
+df_adult_meta2 = df_metadata_maternal_ur.select(["Mother_ID", "Mother_age", "GD_Delivery", "Group", "Mode_birth", "Reject", "Infant_sex"]).rename({"Mother_age": "Age_at_conception"
+, "GD_Delivery": "gestation_day_at_delivery", "Group": "obesity_classification", "Mode_birth": "mode_of_birth"}).clean_names()
+df_adult = pl.concat([df_adult_meta1, df_adult_meta2]).unique().write_csv(folder_path / "data_adult_meta.csv", separator=";") # Concatenate long
+
+df_adult_weight1 = df_metadata_maternal_bl.select(["Mother_ID", "GD_day", "GD_targeted", "Mother_Weight", "BCS"]).rename({"GD_day": "sample_gestation_day"
+, "GD_targeted": "target_gestation_day", "Mother_Weight": "weight_at_gestation_day", "BCS": "body_condition_score"})
+df_adult_weight2 = df_metadata_maternal_ur.select(["Mother_ID", "GD", "Target_GD", "Mother_Weight", "BCS"]).rename({"GD": "sample_gestation_day"
+, "Target_GD": "target_gestation_day", "Mother_Weight": "weight_at_gestation_day", "BCS": "body_condition_score"})
+df_adult_weight = pl.concat([df_adult_weight1, df_adult_weight2]).unique().write_csv(folder_path / "data_adult_weight.csv", separator=";") # Concatenate long
+
+# Create the adult linking tables
+
+# matter1 = "Blood" define above in infant linking tables
+# matter2 = "Urine" define above in infant linking tables
+matter3 = "Placenta"
+
+df_link_adult1 = df_metadata_maternal_bl.select(["Mother_ID", "Exp", "GD_day", "GD_targeted", "Batch", "Dilution_factor"]).rename({"GD_day": "day_sample_taken"
+, "GD_targeted": "target_sampling_day", "Exp": "adult_sample_id"}).with_columns(pl.lit(matter1).alias("type_of_matter"))
+df_link_adult2 = df_metadata_maternal_ur.select(["Mother_ID", "Exp", "GD", "Target_GD", "Batch", "Dilution_factor"]).rename({"GD": "day_sample_taken"
+, "Target_GD": "target_sampling_day", "Exp": "adult_sample_id"}).with_columns(pl.lit(matter2).alias("type_of_matter"))
+df_link_adult3 = df_metadata_maternal_pl.select(["Mother_ID", "Exp", "GD", "Target_GD", "Batch", "Dilution_factor"]).rename({"GD": "day_sample_taken"
+, "Target_GD": "target_sampling_day", "Exp": "adult_sample_id"}).with_columns(pl.lit(matter3).alias("type_of_matter"))
+
+df_adult_link = pl.concat([df_link_adult1, df_link_adult2, df_link_adult3]).write_csv(folder_path / "data_adult_sample_meta.csv", separator=";") # Concatenate long
+
```

Comment:
Same as above

Reply:
Since only one in three files had different headings I renamed that, and then did the full renaming in the concat file.